### PR TITLE
pin a min version of h5py

### PR DESCRIPTION
**Pull request by nabobalis**
I don't think this needs a changelog? 

Fixes https://github.com/sunpy/sunpy/issues/5300

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -67,6 +67,9 @@ net =
   zeep>=3.4.0
 timeseries =
   h5netcdf>=0.8.1
+  # While a not direct dependacy
+  # We need to raise this to ensure the goes netcdf files open.
+  h5py>=3.1.0
```
Comment (nabobalis) [inline setup.cfg:72]: So I wanted to pin to a >2 version but in 3.1 they fixed a bug with opening data with big endian and I felt that was worth pinning to as a min.

File: setup.cfg [hunk 2]
```diff
@@ -67,6 +67,9 @@ net =
   zeep>=3.4.0
 timeseries =
   h5netcdf>=0.8.1
+  # While a not direct dependacy
```
Comment (nabobalis) [inline setup.cfg:70]:
```suggestion
  # While a not direct dependency
```

**Comment by Cadair** [comment]
How old is the 3.1 release ooi?

**Comment by nabobalis** [comment]
> How old is the 3.1 release ooi?

Released: Nov 6, 2020

**Comment by nabobalis** [comment]
This has been manually added to 2.1

**Pull request merged**